core/src/main/scala/filodb.core/memstore/ratelimit/CardinalityStore.scala

```diff
@@ -14,6 +14,16 @@ import java.io.Closeable
  */
 case class Cardinality(prefix: Seq[String], tsCount: Int, activeTsCount: Int, childrenCount: Int, childrenQuota: Int)
 
+/**
+ * Interface for CardinalityStore iterator.
+ */
+trait CardIter extends Iterator[Cardinality] with Closeable {
+  /**
+   * Must be called before iteration.
+   */
+  def seek(): Unit
+}
+
```

Comment:
The intent is to force the client to acknowledge any I/O errors-- the previous setup just ate any errors that occurred while `Cardinality`s were scanned. The client should be more aware of the backend, now, anyways, since `close()` will be left uncalled if the iterator is not exhausted.